biggraphite/accessor.py

```diff
@@ -0,0 +1,456 @@
+        time_start_ms = timestamp_ms - time_offset_ms
+        return (self.__insert_statement, (metric_name, time_start_ms, time_offset_ms, value))
+
+    def insert_points(self, metric_name, timestamps_and_values):
+        """Insert points for a given metric.
+
+        Args:
+          metric_name: A graphite-like metric name (like "my.own.metric")
+          timestamps_and_values: An iterable of (timestamp in seconds, values as double)
+        """
+        assert self.is_connected, "connect() wasn't called"
+        statements_and_args = [
+            self._make_insert_points(metric_name, t, v)
+            for t, v in timestamps_and_values
+        ]
+        c_concurrent.execute_concurrent(
+            self.__session, statements_and_args, concurrency=self.__executor_threads * 50)
+
+    def drop_all_metrics(self):
+        """Delete all metrics from the database."""
+        assert self.is_connected, "connect() wasn't called"
+        self.__session.execute("TRUNCATE metrics;")
+        self.__session.execute("TRUNCATE raw;")
+
+    @staticmethod
+    def _round_down(timestamp, precision):
+        return int(timestamp) // precision * precision
+
+    def _make_all_points_selects(self, metric_name, time_start_ms, time_end_ms):
+        # We fetch with ms precision, even though we only store with second precision.
+        first_row = self._round_down(time_start_ms, self._ROW_SIZE_MS)
+        last_row = self._round_down(time_end_ms, self._ROW_SIZE_MS)
+        res = []
+        # xrange(a,b) does not contain be, so we use last_row+1
+        for row_start_ms in xrange(first_row, last_row + 1, self._ROW_SIZE_MS):
+            row_min_offset = -1  # Selects all
+            row_max_offset = self._ROW_SIZE_MS + 1   # Selects all
+            if row_start_ms == first_row:
+                row_min_offset = time_start_ms - row_start_ms
+            if row_start_ms == last_row:
+                row_max_offset = time_end_ms - row_start_ms
+            statement = (self.__select_statement,
+                         (metric_name, row_start_ms, row_min_offset, row_max_offset))
+            res.append(statement)
+        return res
+
+    def update_metric(self, metric_metadata):
+        """Update a metric definition from a MetricMetadata."""
+        components = self._components_from_name(metric_metadata.name)
+        empty_components = _COMPONENTS_MAX_LEN - len(components)
+        if empty_components > 0:
+            components += [None] * empty_components
+        statement_args = [metric_metadata.name, metric_metadata.as_string_dict()] + components
+        self.__session.execute(self.__insert_metrics_statement, statement_args)
+
+    @staticmethod
+    def median_aggregator(coverage, points):
+        """The default aggregator for fetch(), returns the median of points."""
+        if points:
+            return statistics.median(points)
+        else:
+            return None
+
+    def get_metric(self, metric_name):
```

Comment:
I'm not sure if I should use get_metric or glob_metric_names to list the childs of a directory.